Foundation/NSNumberFormatter.swift

```diff
@@ -27,87 +43,796 @@ public class NSNumberFormatter : NSFormatter {
+//        _setFormatterAttribute(formatter, attributeName: kCFNumberFormatterPaddingPosition, value: _paddingPosition.rawValue._bridgeToObject()._cfObject)
+//        _setFormatterAttribute(formatter, attributeName: kCFNumberFormatterPaddingCharacter, value: _paddingCharacter?._cfObject)
+//        _setFormatterAttribute(formatter, attributeName: kCFNumberFormatterMultiplier, value: _multiplier?._cfObject)
+//        _setFormatterAttribute(formatter, attributeName: kCFNumberFormatterPositivePrefix, value: _positivePrefix?._cfObject)
+//        _setFormatterAttribute(formatter, attributeName: kCFNumberFormatterPositiveSuffix, value: _positiveSuffix?._cfObject)
+//        _setFormatterAttribute(formatter, attributeName: kCFNumberFormatterNegativePrefix, value: _negativePrefix?._cfObject)
+//        _setFormatterAttribute(formatter, attributeName: kCFNumberFormatterNegativeSuffix, value: _negativeSuffix?._cfObject)
+//        _setFormatterAttribute(formatter, attributeName: kCFNumberFormatterPerMillSymbol, value: _percentSymbol?._cfObject)
+//        _setFormatterAttribute(formatter, attributeName: kCFNumberFormatterInternationalCurrencySymbol, value: _internationalCurrencySymbol?._cfObject)
+//        _setFormatterAttribute(formatter, attributeName: kCFNumberFormatterCurrencyGroupingSeparator, value: _currencyGroupingSeparator?._cfObject)
+//        _setFormatterAttribute(formatter, attributeName: kCFNumberFormatterIsLenient, value: _lenient._bridgeToObject()._cfObject)
+        _setFormatterAttribute(formatter, attributeName: kCFNumberFormatterUseSignificantDigits, value: _usesSignificantDigits._cfObject)
+        
+        
+        if _minimumIntegerDigits <= 0 && _minimumFractionDigits <= 0 {
+            _setFormatterAttribute(formatter, attributeName: kCFNumberFormatterMinSignificantDigits, value: _minimumSignificantDigits._bridgeToObject()._cfObject)
+        }
+//        _setFormatterAttribute(formatter, attributeName: kCFNumberFormatterMaxSignificantDigits, value: _maximumSignificantDigits._bridgeToObject()._cfObject)
+    }
     
-    public class func localizedStringFromNumber(num: NSNumber, numberStyle nstyle: NSNumberFormatterStyle) -> String { NSUnimplemented() }
+    internal func _setFormatterAttribute(formatter: CFNumberFormatterRef, attributeName: CFString, value: AnyObject?) {
+        if let value = value {
+            CFNumberFormatterSetProperty(formatter, attributeName, value)
+        }
+    }
     
     // Attributes of an NSNumberFormatter
+    internal var _numberStyle: NSNumberFormatterStyle = .NoStyle
+    public var numberStyle: NSNumberFormatterStyle {
+        get {
+            return _numberStyle
+        }
+        
+        set {
+            _reset()
+            _numberStyle = newValue
+        }
+    }
+    
+    internal var _locale: NSLocale = NSLocale.currentLocale()
+    /*@NSCopying*/ public var locale: NSLocale! {
```

Comment:
That would be nice!
But I think that on a next step we should do something like this:
```swift
    internal var _numberStyle: NSNumberFormatterStyle = .NoStyle
    public var numberStyle: NSNumberFormatterStyle {
        get {
            let rawValue = CFNumberFormatterGetStyle(_cfObject).rawValue
            _numberStyle = NSNumberFormatterStyle(rawValue: UInt(rawValue))!
            return _numberStyle
        }
        
        set {
            _reset()
            _numberStyle = newValue
        }
    }
```
And then we would be sure that both our object and the internal CoreFoundation formatter are in sync, which currently isn't happening. And we could also take leverage of CoreFoundation number formatter default values.

I really dislike the interval vars too, but I can't think of some other option. :disappointed:
Do you see any other way that allows us to keep both worlds in sync?